src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel_Utilities.cs

```diff
@@ -23,43 +25,37 @@ namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline;
 using LspDocumentSymbol = DocumentSymbol;
 using Range = Roslyn.LanguageServer.Protocol.Range;
 
-internal delegate Task<ManualInvocationResponse?> LanguageServiceBrokerCallback(
-    ITextBuffer textBuffer, Func<JToken, bool> capabilitiesFilter, string languageServerName, string method, Func<ITextSnapshot, JToken> parameterFactory, CancellationToken cancellationToken);
+internal delegate Task<TResponse?> LanguageServiceBrokerCallback<TRequest, TResponse>(Request<TRequest, TResponse> request, CancellationToken cancellationToken);
 
 internal sealed partial class DocumentOutlineViewModel
 {
     /// <summary>
     /// Makes an LSP document symbol request and returns the response and the text snapshot used at 
     /// the time the LSP client sends the request to the server.
     /// </summary>
-    public static async Task<(JToken response, ITextSnapshot snapshot)?> DocumentSymbolsRequestAsync(
+    public static async Task<(DocumentSymbolNewtonsoft.NewtonsoftRoslynDocumentSymbol[] response, ITextSnapshot snapshot)?> DocumentSymbolsRequestAsync(
         ITextBuffer textBuffer,
-        LanguageServiceBrokerCallback callbackAsync,
+        LanguageServiceBrokerCallback<DocumentSymbolNewtonsoft.NewtonsoftRoslynDocumentSymbolParams, DocumentSymbolNewtonsoft.NewtonsoftRoslynDocumentSymbol[]> callbackAsync,
         string textViewFilePath,
         CancellationToken cancellationToken)
     {
         ITextSnapshot? requestSnapshot = null;
-        JToken ParameterFunction(ITextSnapshot snapshot)
+
+        var request = new DocumentRequest<DocumentSymbolNewtonsoft.NewtonsoftRoslynDocumentSymbolParams, DocumentSymbolNewtonsoft.NewtonsoftRoslynDocumentSymbol[]>()
```

Comment:
Even though we switched to the new LSP client API that uses concrete types instead of JTOken - the streamjsonrpc instance on the client side is still newtonsoft.  Therefore we have to use type definitions compatible with newtonsoft serialization